### PR TITLE
feature/complex-math

### PR DESCRIPTION
**Pull request by builtbywill**

### CODE REVIEW

File: LATEX.g4 [hunk 1]
```diff
@@ -442,33 +478,34 @@ func_normal_functions_single_arg:
...
+func_cmd_multi_arg:
     FUNC_GCD | FUNC_LCM | FUNC_MAX | FUNC_MIN;
 
-func_operator_names_single_arg:
+func_name_single_arg:
     FUNC_ARSINH_NAME | FUNC_ARCOSH_NAME | FUNC_ARTANH_NAME
     | FUNC_ARCSINH_NAME | FUNC_ARCCOSH_NAME | FUNC_ARCTANH_NAME
-    | FUNC_FLOOR_NAME | FUNC_CEIL_NAME;
+    | FUNC_FLOOR_NAME | FUNC_CEIL_NAME
+    | FUNC_RE_NAME | FUNC_IM_NAME | FUNC_ARG_NAME | FUNC_ABS_NAME;
```
Comment (builtbywill) [inline LATEX.g4:488]: In addition to adding the new function names here, I did some re-naming and organization of the func rules, similar to what I did in the experimental unknown functions branch

File: sandbox/sandbox_complex_math.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+Efphase = compute_expr(Efphase_expr)
+print('Efphase =', Efphase, ', evalf() =>', N(Efphase, 15), ', expected = 1.62312006884331')
+
+# $VLLNL=sqrt(3)*$Efmag;
+# $VLLNL=10552.2774509271
+VLLNL_expr = process_sympy('\\sqrt{3}*\\variable{Efmag}', variable_values=variable_values | {'Efmag': Efmag})
+VLLNL = compute_expr(VLLNL_expr)
+print('VLLNL =', VLLNL, ', evalf() =>', N(VLLNL, 15), ', expected = 10552.2774509271')
+
+# New Claculatins for Skill Builder 3.1.9
```
Comment (jkdufair) [inline sandbox/sandbox_complex_math.py:182]: I know you left this on purpose, @builtbywill 😂

File: tests/complex_test.py [hunk 3]
```diff
@@ -1,21 +1,94 @@
...
-    assert_equal("e^{I\\pi}", Integer(-1))
 
+complex_number_rectangular_examples = [
+    ('3+\\imaginaryI 4', Add(3, Mul(I, 4, evaluate=False), evaluate=False), {}),
+    ('-3-4\\imaginaryI ', Add(-3, Mul(-1, 4, I, evaluate=False), evaluate=False), {}),
+    ('3.14+\\imaginaryI 4.657', Add(Rational(157, 50), Mul(I, Rational(4657, 1000), evaluate=False), evaluate=False), {}),
+    ('3+\\imaginaryJ 4', Add(3, Mul(I, 4, evaluate=False), evaluate=False), {}),
+    ('-3-4\\imaginaryJ ', Add(-3, Mul(-1, 4, I, evaluate=False), evaluate=False), {}),
+    ('3.14+\\imaginaryJ 4.657', Add(Rational(157, 50), Mul(I, Rational(4657, 1000), evaluate=False), evaluate=False), {}),
+    ('\\variable{a}+\\imaginaryJ \\variable{b}', Add(Symbol('a' + hashlib.md5('a'.encode()).hexdigest(), real=True), Mul(I, Symbol('b' + hashlib.md5('b'.encode()).hexdigest(), real=True), evaluate=False), evaluate=False), {}),
```
Comment (jkdufair) [inline tests/complex_test.py:33]: @builtbywill is this MD5 stuff just some internal sympy artifact? Or?

Comment (builtbywill) [reply tests/complex_test.py:33]: @jkdufair no that's just how variable symbol names are generated in the latex2sympy code

Comment (builtbywill) [reply tests/complex_test.py:33]: to differentiate them from just letter symbols

File: src/latex2sympy/latex2sympy.py [hunk 4]
```diff
@@ -673,51 +700,15 @@ def convert_func(self, func):
...
                     name = 'a' + name
                     should_pow = False
                 expr = getattr(sympy.functions, name)(arg, evaluate=False)
 
-            if func_pow and should_pow:
-                expr = sympy.Pow(expr, func_pow, evaluate=False)
-
-            return expr
-
-        elif 'func_normal_multi_arg' in func:
```
Comment (builtbywill) [inline src/latex2sympy/latex2sympy.py:687]: I was able to consolidate some duplicate logic (again, partly pulled from my unknown functions branch) to combine the logic for single arg and multi arg functions

**Review by jkdufair — APPROVED** [review]
OK - having gone through the complex number stuff with you already, it all looks good. And I think the tests do cover all of our use cases (questions about π notwithstanding). Function stuff looks good too. Tests all pass. :thumps:

**Pull request merged**